### DISCUSSION TITLE
URI redirecting (deep link mobile app) not working

### DISCUSSION THREAD

**Discussion by Echelpoel** [vercel/community#2277] [Help] [answered]
### Summary

In `vercel.json` I've set up the following redirect. The destination string is a mobile deep link with a URI.

```
"redirects": [
    { "source": "/redirect-mobile-app/:path*", "destination": "mobile-app://:path*" },
  ]
```

This results in the page redirecting to for example `my-mobile-app:/user/1234/*`. As you can see there is only 1 slash where there should be 2, and also the redirect adds a slash with an asterisk to the end of the redirect url.

I've tried escaping some characters with `\\` as noted in the docs (although the docs don't mention escaping in the destination string, only in the source string) but that triggers a Vercel internal server error.

Any thoughts on this? Thanks in advance!

**Comment by amyegan** [comment]
Hi @Echelpoel! Can you share an example of an expected source path and the destination for its redirect? I want to be sure that I understand your intent so I can make an appropriate suggestion

**Reply by Echelpoel** [reply]
Hi @amyegan, here you go:

Expected source path: `https://redirect-mobile-app.vercel.app/user/1234`
Expected destination path: `mobile-app://user/1234`

Actual destination path: `mobile-app:/user/1234/*`

**Reply by imranmd** [reply]
I am also facing the same problem  
https://islamicdreambook.com/dream-meaning-search/1/

It is working in desktop as expected, but not working in mobile

Here is my vercel json
{
  "trailingSlash": true,
  "redirects": [
    {
      "source": "/dream-meaning-search/1/",
      "destination": "/dream-interpretation-search/1/",
      "statusCode": 301
    },
}

I tried adding
   "has": [
        {
          "type": "header",
          "key": "User-Agent",
          "value": "Mobile"
        }
      ],

**Reply by imranmd** [reply]
One more information that might help you, If I add User-Agent: Mobile  and hit the same url in Postman it is not redirecting, If I remove it it is working

**Reply by imranmd** [reply]
This should be even more useful, I just deleted my project, https://islamicdreambook.com/dream-meaning-search/1/

and on desktop it is breaking, but on mobile it is not, Something is wrong with mobile as user agent in Vercel cache server

**Reply by imranmd** [reply]
I had to take a dirty way to fix it, Deleted domain and added back, but seems like there is caching issue for mobile requests

**Comment by omakise** [comment]
I also met the same problem. Is there any solution?

**Reply by Echelpoel** [reply, marked as answer]
My solution was to create a Next app with an API route that redirects to the given path.

```
import type { NextApiRequest, NextApiResponse } from 'next'

export default function handler(req: NextApiRequest, res: NextApiResponse) {
  const { slug }: any = req.query
  const path = slug?.join('/') || ''

  res.writeHead(301, { Location: `mobile-app://${path}` });

  return res.status(200).end()
}
```

**Comment by jedashford** [comment]
We have this same problem! Trying to redirect to: `sms://18885555555?&body=this%20test` and vercel strips the slashes to a single 'sms:/' no matter what we do to: `sms:/18885555555?&body=this%20test`

We're trying to automate these and do not want to run or download the rest of the mobile application or take on middleware costs and complexity. How can we fix the redirect functionality to return the url we want to return?
